ui/src/views/Tasks/TaskGroup/index.jsx

```diff
@@ -164,16 +169,98 @@ export default class TaskGroup extends Component {
     searchTerm: null,
   };
 
+  static unsubscribe() {
+    if (!TaskGroup.listener) {
+      return;
+    }
+
+    TaskGroup.listener.unsubscribe();
+    TaskGroup.listener = null;
+  }
+
+  static subscribe({ taskGroupId, subscribeToMore }) {
+    if (TaskGroup.listener && TaskGroup.listener.taskGroupId === taskGroupId) {
+      return TaskGroup.listener;
+    }
+
+    if (TaskGroup.listener && TaskGroup.listener.taskGroupId !== taskGroupId) {
+      TaskGroup.unsubscribe();
+    }
+
+    const unsubscribe = subscribeToMore({
+      document: taskGroupSubscription,
+      variables: {
+        taskGroupId,
+        subscriptions: [
+          'tasksDefined',
+          'tasksPending',
+          'tasksRunning',
+          'tasksCompleted',
+          'tasksFailed',
+          'tasksException',
+        ],
+      },
+      updateQuery(previousResult, { subscriptionData }) {
+        const { tasksSubscriptions } = subscriptionData.data;
```

Comment:
I think `tasksSubscriptions` has a single task status in it?  The doubly-plural variable name suggests otherwise, so I'm not sure.

Reply:
This is a similar reasoning as https://github.com/taskcluster/taskcluster-web-server/pull/49#issuecomment-429380389. We are subscribing to tasks, not the return value.